RealEstateDataCleaning.py

```diff
@@ -8,113 +8,160 @@
 import pandas as pd
 import numpy as np
 
+# aea: argparse here
 input_file = 'Real Estate Model - Candler Park.csv'
 output_file = 'cleaned_data_cp5.csv'
-#lists of keywords to make dummy vars of if they are present in an observation's description  
+
+
+# aea: better as a map features = { 'Exterior' : [...], 'BasementDesc' : [...], ... }
+# aea: this solves capitalization as well (mention PEP8)
+# lists of keywords to make dummy vars of if they are present in an observation's description
 Exterior_features = ['Porch','Patio','Deck','Fenced','Out-Building']
 BasementDesc_features = ['Crawl Space','Slab', 'Basement']
 ParkingDesc_features = ['Driveway','Carport', '1 Car Garage', '2 Car Garage', '3 Car Garage']
 Stories_features = ['1 Story', '2 Story', 'More']
 
 def make_YearBuilt_bins(df, size):
-	'''Divides YearBuilt into bins and creates new column named YearBuiltBins
+    """Divides YearBuilt into bins and creates new column named YearBuiltBins
 
     Args:
-        df (pandas DataFrame): The DataFrame to make bins on 
+        df (pandas DataFrame): The DataFrame to make bins on
         size (int): The size of each bin
 
     Returns:
         pandas DataFrame: with an additional column named YearBuiltBins
-    '''
-
-	bin_cutoffs = df['YearBuilt'].min()
-	bins = []
-	while bin_cutoffs < df['YearBuilt'].max()+size:
-	    bins.append(bin_cutoffs)
-	    bin_cutoffs += size
-	df['YearBuiltBins'] = pd.cut(df['YearBuilt'], bins)
-	return df
+    """
+
+    bin_cutoffs = df['YearBuilt'].min()
```

Comment:
These are all tab to space changes.  I am personally a space guy.  PEP8 prefers them as well.  The import bit is that there was a mix of tab and spaces indentation in this script, which is undesirable.

```
def because():
   if you_have_a_mix:
               then some things
               look like
               this
    else:
        and
        some look like
        this